python/torch_mlir_e2e_test/torchscript/framework.py

```diff
@@ -238,40 +269,89 @@ def generate_golden_trace(test: Test) -> Trace:
-    """Invoke the given `Test`'s with the provided `TestConfig`."""
-    results = []
-    for test in tests:
-        # TODO: Precompile everything in parallel.
-        try:
-            golden_trace = generate_golden_trace(test)
-            compiled = config.compile(test.program_factory())
-        except Exception as e:
-            results.append(
-                TestResult(unique_name=test.unique_name,
-                           compilation_error="".join(traceback.format_exception(
-                               type(e), e, e.__traceback__)),
-                           runtime_error=None,
-                           trace=None,
-                           golden_trace=None))
-            continue
-        # TODO: Run in parallel.
-        try:
-            trace = config.run(compiled, golden_trace)
-        except Exception as e:
-            results.append(
-                TestResult(unique_name=test.unique_name,
-                           compilation_error=None,
-                           runtime_error="".join(traceback.format_exception(
-                               type(e), e, e.__traceback__)),
-                           trace=None,
-                           golden_trace=None))
-            continue
+def compile_and_run_test(test_name: str, results: mp.Queue, tests: Dict[str, Test],
+                 config: TestConfig) -> Any:
+    test = tests[test_name]
 
+    try:
+        golden_trace = generate_golden_trace(test)
+        compiled = config.compile(test.program_factory())
+    except Exception as e:
         results.append(
             TestResult(unique_name=test.unique_name,
-                       compilation_error=None,
+                       compilation_error="".join(
+                           traceback.format_exception(type(e), e,
+                                                      e.__traceback__)),
                        runtime_error=None,
-                       trace=trace,
-                       golden_trace=golden_trace))
+                       trace=None,
+                       golden_trace=None))
+        return
+    try:
+        trace = config.run(compiled, golden_trace)
+    except Exception as e:
+        results.append(
+            TestResult(unique_name=test.unique_name,
+                       compilation_error=None,
+                       runtime_error="".join(
+                           traceback.format_exception(type(e), e,
+                                                      e.__traceback__)),
+                       trace=None,
+                       golden_trace=None))
+        return
+
+    results.append(TestResult(unique_name=test.unique_name,
+                        compilation_error=None,
+                        runtime_error=None,
+                        trace=clone_trace(trace),
+                        golden_trace=golden_trace))
+
+
+def run_tasks_in_parrallel(task_queue: mp.Queue, task):
+    NUMBER_OF_PROCESSES = min(mp.cpu_count(), task_queue.qsize())
```

Comment:
Usually it is good to have slightly more processes than CPU's. So maybe something like `int(mp.cpu_count() * 1.1)` or something.